universe/Universe.cpp

```diff
@@ -1660,6 +1660,21 @@ void Universe::ForgetKnownObject(int empire_id, int object_id) {
     objects.Remove(object_id);
 }
 
+void Universe::CheckShipDesignVisibility(int empire_id, int object_id, Visibility vis) {
```

Comment:
I don't like a name that suggests it unconditionally gives visibility for a particular ship design (of the specified ship ID) to a particular empire, when it actually only does so if the specified ship is visible above some threshold.